scripts/o3de/tests/test_repo_properties.py

```diff
@@ -0,0 +1,180 @@
+                         None, None, None, None,
+                         'project1 project2 project3', 'project2', None, [{'project_name':'project1'}, {'project_name':'project3'}],
+                         None, None, None, None,
+                         0),
+            # test replace a project
+            pytest.param(pathlib.Path('F:/repotest'),'Repotest1', 
+                         None, None, None, None,
+                         'project1 project2', None, 'project3', [{'project_name':'project3'}],
+                         None, None, None, None,
+                         0),
+            # test remove project with a empty parameter
+            pytest.param(pathlib.Path('F:/repotest'),'Repotest1', 
+                         None, None, None, None,
+                         'project1 project2', '', None, [{'project_name':'project1'}, {'project_name':'project2'}],
+                         None, None, None, None,
+                         0),
+            # test add template
+            pytest.param(pathlib.Path('F:/repotest'),'Repotest1', 
+                         None, None, None, None,
+                         None, None, None, None,
+                         'template1 template2 template3', None, None, [{'template_name':'template1'}, {'template_name':'template2'}, 
+                                                                       {'template_name':'template3'}],
+                         0),
+            # test remove multiple templates
+            pytest.param(pathlib.Path('F:/repotest'),'Repotest1', 
+                         None, None, None, None,
+                         None, None, None, None,
+                         'template1 template2 template3', 'template1 template2', None, [{'template_name':'template3'}],
+                         0),
+            # test replace a temlpate
+            pytest.param(pathlib.Path('F:/repotest'),'Repotest1', 
+                         None, None, None, None,
+                         None, None, None, None,
+                         'template1 template2 template3', None, "template48", [{'template_name':'template48'}],
+                         0),
+            # test add all objects
+            pytest.param(pathlib.Path('F:/repotest'),'Repotest1', 
+                         'gem1', None, None, [{'gem_name':'gem1'}],
+                         'project1', None, None, [{'project_name':'project1'}],
+                         'template1 template2', None, None, [{'template_name':'template1'}, {'template_name':'template2'}],
+                         0),
+            # test remove all object types
+            pytest.param(pathlib.Path('F:/repotest'),'Repotest1', 
+                         'gem1 gem2', 'gem1', None, [{'gem_name':'gem2'}],
+                         'project1', 'project1', None, [],
+                         'template1 template2 template3', 'template1 template3', None, [{'template_name':'template2'}],
+                         0)
+            ]
```

Comment:
We need to test adding multiple versions of gems and removing gems by version.  I recommend making a new test in this file specifically for version testing because it'll likely be more complex.  It should test:
1. Adding a gem with version 1.0.0 and then the same gem with version 2.0.0
2. Adding a gem with version 1.0.0 and then the same gem with version 2.0.0 and then removing version 2.0.0
3. Adding a gem with version 1.0.0 and then the same gem with version 2.0.0 and then removing version 1.0.0
4. Adding a gem with version 1.0.0 and then the same gem with version 2.0.0 and then removing version 1.0.0 and 2.0.0
5. Adding a gem with version 1.0.0 and then the same gem with the same version but other updated gem.json fields
6. Adding a gem with version 1.0.0 and then the same gem with version 2.0.0 and then adding version 2.0.0 with updated gem.json fields
7. Adding a gem with multiple versions and then replacing all gems with another set of gems

We will likely also need to add tests for users who have gems that don't contain a version field in their gem.json because they're older gems.  Something like:
1. Adding a gem without a version, then adding a gem with the same name but with a version
2. Adding a gem without a version, then adding a gem with the same name but with a version and then removing the unversioned gem by using the delete_gems = ['gemA==0.0.0']  or delete_gems = ['gemA<1.0.0']